### PR TITLE
update readme for promise and server push

### PR DESCRIPTION
**Pull request by georgeu2000**
Issue https://github.com/igrigorik/http-2/issues/58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -246,9 +246,10 @@ conn.on(:stream) do |stream|
     }
 
     # initiate server push stream
+    push_stream = nil
     stream.promise(head) do |push|
-      push.headers({ ... })
-      push.data(...)
```
Comment (georgeu2000) [inline README.md]: It seems like we should send the data later.

Comment (igrigorik) [inline README.md]: That's what .promise does above.. The latter call is to demonstrate that you can send additional headers if you want, before you proceed with body.

**Comment by igrigorik** [comment]
LGTM, modulo the .headers({...}) change; we shouldn't send same headers data twice.

File: README.md (hunk 1 above)
Comment (georgeu2000) [inline README.md]: In my testing, this sends the data immediately. I don't see how it sends it later.

```ruby
stream.promise(head) do |push|
       push.headers({ ... })
      push.data(...)
```

Comment (igrigorik) [inline README.md]: Sorry about the confusion, I meant that we should keep .headers({...}) in place to demonstrate how to send additional headers. the .data() calls can happen in the callback or later -- this is up to the application to decide based on what it's pushing, etc.

Comment (georgeu2000) [inline README.md]: Oh, I see.

**Comment by igrigorik** [comment]
Scratching my head as to why Travis is failing all of the sudden.. but we'll deal with that separately.

Looks good now, thanks!

**Pull request merged**